packages/eslint-plugin/src/rules/no-floating-promises.ts

```diff
@@ -296,12 +325,39 @@ export default createRule<Options, MessageId>({
   },
 });
 
+function isPromiseArray(checker: ts.TypeChecker, node: ts.Node): boolean {
+  const type = checker.getTypeAtLocation(node);
+  for (const ty of tsutils
+    .unionTypeParts(type)
+    .map(t => checker.getApparentType(t))) {
+    if (checker.isArrayType(ty)) {
+      const arrayType = checker.getTypeArguments(ty)[0];
+      if (isPromiseLike(checker, node, arrayType)) {
+        return true;
+      }
```

Comment:
[Style] Just nitpicking, the type (at least in the current tests) is only ever up to one of these two possibilities (arrays and tuples). So returning early is doable.

```ts
if (checker.isArrayType(ty)) {
  return isPromiseLike(checker, node, checker.getTypeArguments(ty)[0]);
}
```

Reply:
Could you clarify this? I'm not following, sorry.